src/Core/Aetherling/Passes/Timing.hs

```diff
@@ -0,0 +1,265 @@
+module Aetherling.Passes.Timing (
+  distributeUtil,
+  RetimeComposeParPolicy,
+  rcpDefault,
+  rcpRetimeReadyValid,
+  retimeComposePar,
+  retimeComposePar'
+) where
+import Aetherling.Operations.AST
+import Aetherling.Operations.Types
+import Aetherling.Operations.Ops
+import Aetherling.Operations.Compose
+import Aetherling.Analysis.Latency
+import Aetherling.Analysis.PortsAndThroughput
+import Aetherling.Passes.MapOps
+import Data.Ratio
+import Data.List
+
+-- | Given an AST, distribute all LogicalUtil ops to leaf ops, so that
+-- each leaf op in the output AST is wrapped (if needed) with a
+-- LogicalUtil whose ratio is the product of the LogicalUtil ratios on
+-- the path from the leaf to the root (except at ReadyValid
+-- boundaries, the ratio is reset to 1). This allows us to see the
+-- real speed of an op without inspecting the whole AST. Named by
+-- analogy to mathematical distribution A(x+y) -> Ax+Ay. (A is the
+-- metaphorical LogicalUtil, x and y the ops).
+--
+-- Note: This will not remove Delay ops, but their delay amount will
```

Comment:
Add to this comment explaining that retimeComposePar can be run after this op to solve for the correct delay amounts. Thus, setting them all to 1 in distirbuteUtil not a problem